### DISCUSSION TITLE
What is the best way to install?

### DISCUSSION THREAD

**Discussion by Timer2000** [BJReplay/ha-solcast-solar#15] [General]
Hello,

Very good that new Integration.
Thank you for this.

Can someone write the best Way for Upgrade? I was using the original Version before and want to use your Version. What are the Steps for Changing?

**Comment by phrozen77** [comment]
Got the same question :)

**Comment by BJReplay** [comment]
See the readme - https://github.com/BJReplay/ha-solcast-solar?tab=readme-ov-file#manually-in-hacs

Until it is added back into HACS, manually in HACS.

**Reply by phrozen77** [reply]
Yeah, that much i understood :) 

The question, for me at least (i suppose for Timer2000 as well) was on how to do an as seamless as possible upgrade from oziee/ha-solcast-solar to this integration, if possible without loosing old data and the need to reconfigure everything.

**Comment by NongJoWo** [comment]
Same question! Currently the old version is still running in my HA. Do I need to remove it completely first or can I overwrite the existing folder "solcast_solar" with new files and restart HA to load the new integration? Many thanks in advance...

**Comment by lamixer** [comment]
I just now did this, so might be too early to say for sure ... but:

I waited until HA bugged me about the Ozzie version being removed from HACS, it asked me if I wanted to remove or ignore. I left it with no answer and opened a new window to HA to go to the integration and make a note of my damping settings and my API key. 

Then I returned to the question and chose to allow HACS to remove the old version from my system.

At that point, if I went to my Solcast overview page, there was no data showing.

Then I installed the BJReplay fork as directed (I never saw an 'INSTALL' button, only DOWNLOAD). After a restart of HA the integration was there and my API key and damping was all still there. Going to my Solcast overview page the days of solar projection was back.

So, worked pretty much seamlessly for me! I hope it does for you also.

**Comment by autoSteve** [comment]
Confirming correct method (at the moment, until it's in the HACS store this is installing a custom repository):

- Remove oziee/ha-solcast-solar in HACS (I had this in the Custom Repositories list)
- Add custom repository (three verical dots menu, top right) ```https://github.com/BJReplay/ha-solcast-solar``` as an ```Integration```
- Search Solcast, click on it, which will show the readme, with a 'download' button bottom right
- Download, which will replace the code at /homeassistant/custom_components/solcast_solar
- Restart HA
- Job done

History/config/all else should remain.

**Reply by phrozen77** [reply]
Seems to have worked - thank you!

**Reply by autoSteve** [reply]
Too easy.

**Comment by NongJoWo** [comment]
Successfully migrated... Many thanks!

**Comment by autoSteve** [comment]
> Can someone write the best Way for Upgrade? I was using the original Version before and want to use your Version. What are the Steps for Changing?

See my post re. changeover in this discussion, @Timer2000. It's super simple, amd won't result in data loss.

**Reply by Timer2000** [reply]
Yes, thank you. I will try it tomorow

**Comment by chalkstorm** [comment]
I've just made the change after the HA core update killed my Solcast readings!  Followed the instructions above and boom, it works.  Thanks to BJRelay for providing this.

**Comment by grillsuder** [comment]
Installed it successful after I deinstalled the old integration but all entities show "0". What can I do?

**Reply by chalkstorm** [reply]
I reloaded the data and it came through (was zero initially)

**Reply by autoSteve** [reply]
The Solcast API probably was busy. I was going to respond, "Look at the logs..."

**Comment by speedster-pta** [comment]
Thanks for everyone's assistance on this. I've been using oziee's integration for a good while but that's obviously been nerfed. I tried switching to this one as outlined in the steps above but after downloading and reinstalling I got a migration error. I then deleted the integration and re-downloaded it and now I just get a "failed to set-up" error. Any idea what I'm doing wrong?

**Reply by BJReplay** [reply]
Logs, please, but in both cases, probably time outs or 429 errors trying to connect to Solcast while you were migrating.

If you try again - to collect the logs that we need to have any hope of helping - it will probably work.

Without logs, we can't guess.

Gather your logs, and if it doesn't work, you should have errors.

If they aren't ones mentioned in the pinned discussions, please open an issue.

**Reply by speedster-pta** [reply]
A Home Assistant update solved my issue.
